cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -306,4 +341,36 @@ private void publishStopProgram(String stopPayload) {
     programStateWriter.killed(programRunId);
     LOG.debug("Published program stop message for run {}", programRunInfo.getRun());
   }
+
+  /**
+   * Add notification to map if associated by a tethered peer
+   */
+  private Map<String, List<Notification>> getPeerProgramUpdates() {
+    Map<String, List<Notification>> peerToNotifications = new HashMap<>();
+    try (CloseableIterator<Message> iterator = messageFetcher.fetch(NamespaceId.SYSTEM.getNamespace(),
+                                                                    programUpdateTopic,
+                                                                    programUpdateFetchSize,
+                                                                    lastProgramUpdateMessageId)) {
+      if (iterator.hasNext()) {
```

Comment:
should loop through the iterator, otherwise it's sending max 1 notification per call.

Reply:
woops I missed that! nice catch